core/src/package.rs

```diff
@@ -45,6 +81,259 @@ pub enum PackageImplementation {
+#[derive(Clone, Debug, PartialEq)]
+pub enum ArgValue {
+    String(String),
+    Integer(i64),
+    UnsignedInteger(u64),
+    Float(f64),
+    EnumVariant(String),
+}
+
+impl From<ArgValue> for Value {
+    fn from(value: ArgValue) -> Self {
+        match value {
+            ArgValue::String(s) | ArgValue::EnumVariant(s) => Value::from(s),
+            ArgValue::Integer(i) => Value::from(i),
+            ArgValue::UnsignedInteger(ui) => Value::from(ui),
+            ArgValue::Float(f) => Value::from(f),
+        }
+    }
+}
+
+// We implement Into<String> since we want to consume the value
+impl From<ArgValue> for String {
+    fn from(value: ArgValue) -> Self {
+        match value {
+            ArgValue::String(s) | ArgValue::EnumVariant(s) => s,
+            ArgValue::Integer(i) => format!("{i}"),
+            ArgValue::UnsignedInteger(ui) => format!("{ui}"),
+            ArgValue::Float(f) => format!("{f}"),
+        }
+    }
+}
+
+impl ArgValue {
+    pub fn get_type(&self) -> ArgType {
+        match &self {
+            ArgValue::String(_) => PrimitiveArgType::String.into(),
+            ArgValue::Integer(_) => PrimitiveArgType::Integer.into(),
+            ArgValue::UnsignedInteger(_) => PrimitiveArgType::UnsignedInteger.into(),
+            ArgValue::Float(_) => PrimitiveArgType::Float.into(),
+            // We have variant-erasure but I think that's OK
+            ArgValue::EnumVariant(_) => ArgType::Enum(vec![]),
+        }
+    }
+
+    pub fn get_enum_variant(self) -> Option<String> {
+        if let ArgValue::EnumVariant(s) = self {
+            Some(s)
+        } else {
+            None
+        }
+    }
+
+    pub fn unwrap_enum_variant(self) -> String {
+        self.get_enum_variant().unwrap()
+    }
+
+    pub fn as_string(&self) -> Option<&str> {
+        if let ArgValue::String(s) = &self {
+            Some(s)
+        } else {
+            None
+        }
+    }
+
+    pub fn get_string(self) -> Option<String> {
+        if let ArgValue::String(s) = self {
+            Some(s)
+        } else {
+            None
+        }
+    }
+
+    pub fn unwrap_string(self) -> String {
+        self.get_string().unwrap()
+    }
+
+    pub fn get_integer(self) -> Option<i64> {
+        if let ArgValue::Integer(i) = self {
+            Some(i)
+        } else {
+            None
+        }
+    }
+
+    pub fn unwrap_integer(self) -> i64 {
+        self.get_integer().unwrap()
+    }
+
+    pub fn get_unsigned_integer(self) -> Option<u64> {
+        if let ArgValue::UnsignedInteger(i) = self {
+            Some(i)
+        } else {
+            None
+        }
+    }
+
+    pub fn unwrap_unsigned_integer(self) -> u64 {
+        self.get_unsigned_integer().unwrap()
+    }
+
+    pub fn get_float(self) -> Option<f64> {
+        if let ArgValue::Float(f) = self {
+            Some(f)
+        } else {
+            None
+        }
+    }
+
+    pub fn unwrap_float(self) -> f64 {
```

Comment:
Not sure about this one, but I think that we maybe want to skip these unwrap_XXX functions? Writing `get_float().unwrap()` at the call site is not that much more verbose and it avoids having so many duplicate utility methods.